pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/errorprone/BeanMembersShouldSerializeRule.java

```diff
@@ -81,7 +84,8 @@ public Object visit(ASTClassOrInterfaceDeclaration node, Object data) {
         for (Map.Entry<VariableNameDeclaration, List<NameOccurrence>> entry : vars.entrySet()) {
             VariableNameDeclaration decl = entry.getKey();
             AccessNode accessNodeParent = decl.getAccessNodeParent();
-            if (entry.getValue().isEmpty() || accessNodeParent.isTransient() || accessNodeParent.isStatic()) {
+            if (entry.getValue().isEmpty() || accessNodeParent.isTransient() || accessNodeParent.isStatic() 
+                    || classHasLombok || hasIgnoredAnnotation((Annotatable) accessNodeParent)) {
```

Comment:
rather than checking here for each variable if the class is to be skipped due to lombok annotations, we should probably directly in line 66 do:

```java
if (hasLombokAnnotation(node)) {
  return super.visit(node, data);
}
```